pymagicc/api.py

```diff
@@ -39,38 +39,31 @@ def __init__(self, root_dir=None):
 
         if root_dir is not None:
             self.is_temp = False
-            if not exists(root_dir):
-                makedirs(root_dir)
         else:
             # Create a temp directory
             self.is_temp = True
             self.root_dir = mkdtemp(prefix="pymagicc-")
 
     def __enter__(self):
-        if not self.is_initialised():
-            self.init()
+        if self.is_temp and not exists(self.run_dir):
+            self.create_copy()
```

Comment:
Perhaps there should be a check for when a root_dir is provided but has not already been initialised.

You could either:
* Raise an exception
* Do nothing
* Create call `create_copy` (this is the previous behaviour)

Reply:
You mean for the case when a `root_dir` is defined in

magicc = MAGICC(root_dir="/tmp/magicctest")

but there is actually no MAGICC binary and files in there?

Reply:
That could be useful addition, as I removed the possibility to define a location for a MAGICC package in `pymagicc.run`

Reply:
Since this is the __enter__ method, more specifically:
```
with MAGICC(root_dir="/tmp/magicctest") as magicc:
```

The `is_initialised` check was to conform with option 3.

Reply:
Ok -- then
```python
def __enter__(self):
        if not exists(self.run_dir):
            self.create_copy()
        return self
```
should do it, right?

Reply:
Hm, still unsure what do here, @lewisjared 

The "promise" of using "with" is sort of to always clean up afterwards.
So now, if a user wanted to create a copy out of the MAGICC copy within Pymagicc they would have to
```python
m = MAGICC(root_dir="/tmp/magicctesting")
m.create_copy()
```
This would create a copy and even create "magicctesting" if it doesn't exist.

Is there a need to have the same behaviour with "with" (and keep the copy afterwards)?

Reply:
It seems to me that if people want a copy, they should have to make it outside of the context manager. As Robert says, if you use `with` the promise is that things are cleaned up.

How about:

- if you use `with` alongside `root_dir=`, the `magicc_dir` is cleaned up anyway (i.e. deleted) and a warning is thrown which tells the user that this is happening
- the only way to create a copy is as below

```python
m = MAGICC(root_dir="/tmp/magicctesting")
m.create_copy()
```

this would create the copy as well as the `root_dir` if it doesn't already exist

Reply:
There isn't any promises other than the ones that are documented!
The cases outlined by @znicholls are the most logical. You would reasonably expect that the MAGICC instance exists for the life of the with block and an alternative API is provided if you want a long-lived copy. 

Perhaps we should start adding some logger messages to the library to clarify the actions being taken

Reply:
>if you use with alongside root_dir=, the root_dir is cleaned up anyway (i.e. deleted) and a warning is thrown which tells the user that this is happening

If you had a created a copy of MAGICC, then Pymagicc shouldn't delete it I think.

Maybe 
> Raise an exception
if there is an empty `root_dir`given is the safest option.

We could expose `create_copy` as a general helper, something like:

```python
pymagicc.create_magicc_copy("/tmp/mymagicc-test")
with MAGICC(root_dir="/tmp/mymagicc-test") as m:
    m.run()

# /tmp/mymagicc-test is still around
```